pkg/controller/installation/core_controller.go

```diff
@@ -1198,18 +1196,6 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 			typhaNodeTLS.TrustedBundle.AddCertificates(prometheusClientCert)
 		}
 
-		// es-kube-controllers needs to trust the ESGW certificate. We'll fetch it here and add it to the trusted bundle.
```

Comment:
I have a setup with a management cluster running this operator code, with successfully connected 316 and master managed clusters, as well as a managed cluster running this operator code. I can access kibana from the management cluster. Is there something specific within Kibana I can check for that would indicate whether or not this is causing any failures?

Reply:
We support a few different modes of certificate provisioning in enterprise.

1. All certificates signed by the tigera-operator CA
2. All certificates signed by an external CSR, with a unique certificate per-pod.
3. What we call "BYO" certificates, which means users can provide their own hand-spun certificate for each component.

I suspect the reason this works in your environment is because the default is option (1), which doesn't require we add the individual certs to the trusted bundle because the tigera-operator CA will already be in the bundle.

While I'm not aware of anyone actually doing option (3), this is the mode in which this code is needed - if the user provides a certificate for es-kube-controllers that isn't actually signed by the tigera-operator signer, then we need to add that certificate to the trusted bundle explicitly.

CC @rene-dekker since we've spoken about this a few times in various contexts.... I really want to stop supporting BYO somehow because it's just so confusing.

Reply:
I think we need to continue to support it for the manager-tls secret. I would propose simply un-documenting the rest.

Reply:
Also note that this code being mistakenly removed here is necessary also for certificate issued by the operator before v3.13.

Reply:
Sounds like we should:

- Remove documentation for all but manager-tls
- Leave the code in place, and add a big comment explaining that this is needed because prior to v3.13 secrets were not signed by a single CA and so we need to include each individually.

That said, we can remove this block of code when kube-controllers no longer talks to es-gateway in any capabity, which hopefully will happen in the future.